app/components/Settings/ConnectButton/ConnectButton.react.js

```diff
@@ -12,6 +22,38 @@ export default class ConnectButton extends Component {
         editMode: PropTypes.bool
     }
 
+    /**
+    * Will check whether connection requests are equal to or greater then 400
+    * @returns {boolean} true if connection error
+    */
+    isConnectionError() {
+        return (this.props.connectRequest.status >= 400 || this.props.saveConnectionsRequest.status >= 400);
+    }
+
+    /**
+     * Checks the connection requests to see if status is loading
+     * @returns {boolean} true if status is loading
+     */
+    loadingStatus() {
+        return (isLoading(this.props.connectRequest.status) || isLoading(this.props.saveConnectionsRequest.status));
+    }
+
+    /**
+     * Checks whether the connection request status (HTTP) is greater or equal to 200 and less then 300
+     * @returns {boolean} true if connection request status >=200 and <300
+     */
+    isValidConnection() {
+        return (this.props.connectRequest.status >= 200 && this.props.connectRequest.status < 300);
+    }
+
+    /**
+     * Checks whether connection status is defined
+     * @returns {boolean} true if connection status is invalid
+     */
+    isInvalidConnectionStatus() {
```

Comment:
This one if weird. The function name  implies that it checks whether the connection failed, but the code checks we're still waiting for a response to out connection request.

I also suspect this may be the source of bug #372 (see my comment below; I'll test if I'm right).